libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -228,7 +228,8 @@ string ABIFunctions::cleanupFunction(Type const& _type, bool _revertOnFailure)
 			if (type.numBytes() == 32)
 				templ("body", "cleaned := value");
 			else if (type.numBytes() == 0)
-				templ("body", "cleaned := 0");
+				// This is disallowed in the type system.
```

Comment:
Do we have tests for `abi.encode('')`, `abi.encodePacked('')`, `abi.encode(hex"")` and `abi.encodePacked(hex"")`?

Reply:
Please check if they would trigger this line.

Reply:
Good idea, though `abi.encode*` supposed to return `Array` if anything and this is about `FixedBytes`?

Also it seems that `abi.encode*` uses `utils().abiEncode` with empty target type pointers.

Reply:
`''` has a type of `bytes memory`.

An explicit conversion is needed to have it as a `FixedBytesType`, but there `bytes0` is disallowed.

Will add tests anyway.